src/OrchardCore/OrchardCore.ContentManagement/Handlers/ContentPartHandlerCoordinator.cs

```diff
@@ -575,41 +343,36 @@ public override async Task GetContentItemAspectAsync(ContentItemAspectContext co
                 }
             }
         }
+
         public override async Task ClonedAsync(CloneContentContext context)
         {
             var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(context.ContentItem.ContentType);
             if (contentTypeDefinition == null)
                 return;
-
-            foreach (var typePartDefinition in contentTypeDefinition.Parts)
-            {
-                var partName = typePartDefinition.PartDefinition.Name;
-                var activator = _contentPartFactory.GetTypeActivator(partName);
-                var part = context.ContentItem.Get(activator.Type, typePartDefinition.Name) as ContentPart;
-
-                if (part != null)
-                {
-                    var partHandlers = _contentPartHandlerResolver.GetHandlers(partName);
-                    await partHandlers.InvokeAsync((handler, context, part) => handler.ClonedAsync(context, part), context, part, _logger);
-                }
-            }
+            await InvokeHandlersForParts(contentTypeDefinition, context, (handler, context, part) => handler.ClonedAsync(context, part));
         }
         public override async Task CloningAsync(CloneContentContext context)
         {
             var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(context.ContentItem.ContentType);
             if (contentTypeDefinition == null)
                 return;
+            await InvokeHandlersForParts(contentTypeDefinition, context, (handler, context, part) => handler.CloningAsync(context, part));
+        }
 
+        private async Task InvokeHandlersForParts<TContext>(ContentTypeDefinition contentTypeDefinition, TContext context, Func<IContentPartHandler, TContext, ContentPart, Task> dispatch) where TContext : ContentContextBase
+        {
             foreach (var typePartDefinition in contentTypeDefinition.Parts)
             {
                 var partName = typePartDefinition.PartDefinition.Name;
+                if (string.IsNullOrEmpty(partName))
```

Comment:
I could be wrong, but you may want to log the error and continue instead of throwing exception. I would think this mean that the part does not exists.

Also, please use Framework type instead of language `String.IsNullOrEmpty(partName)`

Reply:
The line afterwards throws the error anyway and it gets logged like in description, i'm just making the error a bit more descriptive. So String instead of string?